examples/data-frames.dx

```diff
@@ -14,102 +15,106 @@ def studentsMissing : (Fin 3)=>{name: String & age:(Maybe Int) & favColor:(Maybe
+  , {name="Hannah",   get_acne=False, red=False, black=True,  white=False, green=False, yellow=False, brown=False, orange=False, pink=True , purple=False}
+  , {name="Matthew",  get_acne=True,  red=False, black=False, white=False, green=False, yellow=False, brown=True , orange=True , pink=False, purple=False}
+  , {name="Hailey",   get_acne=True,  red=False, black=False, white=False, green=False, yellow=False, brown=False, orange=True , pink=False, purple=False}
+  , {name="Nicholas", get_acne=False, red=True , black=False, white=False, green=False, yellow=True , brown=True , orange=False, pink=True , purple=False}
+  ]
+
+def gradebook : (Fin 3)=>{name:String & age:Int & quiz1:Int & quiz2:Int & midterm:Int & quiz3:Int & quiz4:Int & final:Int} =
+  [ {name="Bob"  , age=12, quiz1=8, quiz2=9, midterm=77, quiz3=7, quiz4=9, final=87}
+  , {name="Alice", age=17, quiz1=6, quiz2=8, midterm=88, quiz3=8, quiz4=7, final=85}
+  , {name="Eve"  , age=13, quiz1=7, quiz2=9, midterm=84, quiz3=8, quiz4=8, final=77}
+  ]
+
+def gradebookMissing : (Fin 3)=>{name:String & age:Maybe Int & quiz1:Maybe Int & quiz2:Maybe Int & midterm:Maybe Int & quiz3:Maybe Int & quiz4:Maybe Int & final:Maybe Int} =
+  [ {name="Bob"  , age=Just 12, quiz1=Just 8,  quiz2=Just 9, midterm=Just 77, quiz3=Just 7,  quiz4=Just 9, final=Just 87}
+  , {name="Alice", age=Just 17, quiz1=Just 6,  quiz2=Just 8, midterm=Just 88, quiz3=Nothing, quiz4=Just 7, final=Just 85}
+  , {name="Eve"  , age=Just 13, quiz1=Nothing, quiz2=Just 9, midterm=Just 84, quiz3=Just 8,  quiz4=Just 8, final=Just 77}
+  ]
+
+def gradebookSeq : (Fin 3)=>{name:String & age:Int & quizzes:(Fin 4=>Int) & midterm:Int & final:Int} =
+  [ {name="Bob"  , age=12, quizzes=[8, 9, 7, 9], midterm=77, final=87}
+  , {name="Alice", age=17, quizzes=[6, 8, 8, 7], midterm=88, final=85}
+  , {name="Eve"  , age=13, quizzes=[7, 9, 8, 8], midterm=84, final=77}
+  ]
+
+def gradebookTable : (Fin 3)=>{name:String & age:Int & quizzes:(Fin 4=>{quiznum:Int & grade:Int}) & midterm:Int & final:Int} =
+  [ {name="Bob"  , age=12, quizzes=[ {quiznum=1, grade=8}
+                                   , {quiznum=2, grade=9}
+                                   , {quiznum=3, grade=7}
+                                   , {quiznum=4, grade=9}], midterm=77, final=87}
+  , {name="Alice", age=17, quizzes=[ {quiznum=1, grade=6}
+                                   , {quiznum=2, grade=8}
+                                   , {quiznum=3, grade=8}
+                                   , {quiznum=4, grade=7}], midterm=88, final=85}
+  , {name="Eve"  , age=13, quizzes=[ {quiznum=1, grade=7}
+                                   , {quiznum=2, grade=9}
+                                   , {quiznum=3, grade=8}
+                                   , {quiznum=4, grade=8}], midterm=84, final=77}
+  ]
+
+
+'# Experiment: Indexing by sets
```

Comment:
Can we please move the experiment to the end?

Reply:
Will do.  I was originally planning to do the entire file twice, once with indexing by row number and once indexing by row elements, but indexing by row elements doesn't work yet.